src/sentry/tasks/integrations/github/open_pr_comment.py

```diff
@@ -242,3 +255,117 @@ def get_top_5_issues_by_count_for_file(
+    except Repository.DoesNotExist:
+        logger.info("github.open_pr_comment.repo_missing", extra={"organization_id": org_id})
+        metrics.incr(OPEN_PR_METRICS_BASE.format(key="error"), tags={"type": "missing_repo"})
+        return
+
+    integration = integration_service.get_integration(integration_id=repo.integration_id)
+    if not integration:
+        logger.info("github.open_pr_comment.integration_missing", extra={"organization_id": org_id})
+        metrics.incr(OPEN_PR_METRICS_BASE.format(key="error"), tags={"type": "missing_integration"})
+        return
+
+    installation = integration.get_installation(organization_id=org_id)
+
+    # GitHubAppsClient (GithubClientMixin)
+    # TODO(cathy): create helper function to fetch client for repo
+    client = installation.get_client()
+
+    if not safe_for_comment(gh_client=client, repository=repo, pull_request=pull_request):
+        metrics.incr(
+            OPEN_PR_METRICS_BASE.format(key="error"),
+            tags={"type": "unsafe_for_comment"},
+        )
+        return
+
+    pr_filenames = get_pr_filenames(gh_client=client, repository=repo, pull_request=pull_request)
+
+    # projects and filenames in sentry for each PR filename
+    reverse_codemappings = [
+        get_projects_and_filenames_from_source_file(org_id, pr_filename)
+        for pr_filename in pr_filenames
+    ]
+
+    top_issues_per_file = [
+        get_top_5_issues_by_count_for_file(projects, sentry_filenames)
+        for projects, sentry_filenames in reverse_codemappings
+    ]
+
+    # format tables in the comment
+    issue_table_contents = [
+        get_issue_table_contents(issue_list) for issue_list in top_issues_per_file
+    ]
+    issue_tables = [format_issue_table(pr_filenames[0], issue_table_contents[0])]
+    issue_tables.extend(
+        [
+            format_issue_table(pr_filenames[i], issue_table_contents[i], toggle=True)
+            for i in range(1, len(pr_filenames))
+        ]
+    )
+    comment_body = format_open_pr_comment(issue_tables)
+
+    # list all issues in the comment
+    issue_list = list(itertools.chain.from_iterable(top_issues_per_file))
+    issue_list = [issue["group_id"] for issue in issue_list]
+
+    try:
+        create_or_update_comment(
+            pr_comment=None,
+            client=client,
+            repo=repo,
+            pr_key=pull_request.key,
+            comment_body=comment_body,
+            pullrequest_id=pull_request.id,
+            issue_list=issue_list,
+            comment_type=CommentType.OPEN_PR,
+            metrics_base=OPEN_PR_METRICS_BASE,
+        )
+    except ApiError as e:
+        if e.json:
+            if ISSUE_LOCKED_ERROR_MESSAGE in e.json.get("message", ""):
+                metrics.incr(
+                    OPEN_PR_METRICS_BASE.format(key="error"),
+                    tags={"type": "issue_locked_error"},
+                )
+                return
+
+            elif RATE_LIMITED_MESSAGE in e.json.get("message", ""):
```

Comment:
maybe we should add a `max_retries` to this task in case we are rate-limited? i'm not sure if you can trigger a retry from inside the task

Reply:
Should we even retry if we're rate limited? Where does the current retry logic live?

Reply:
i believe retry only happens if the task fails like an exception is thrown. it's an argument you can add to the decorator